frontend/src/app/(routes)/multisig/components/BroadCastTxn.tsx

```diff
@@ -0,0 +1,197 @@
+import { fromBase64 } from '@cosmjs/encoding';
+import { TxRaw } from 'cosmjs-types/cosmos/tx/v1beta1/tx';
+import React, { useEffect, useState } from 'react';
+import { MultisigTxStatus } from '@/types/enums';
+import { FAILED_TO_BROADCAST_ERROR } from '@/utils/errors';
+
+interface BroadCastTxnProps {
+  txn: Txn;
+  multisigAccount: MultisigAccount;
+  chainID: string;
+}
+
+const BroadCastTxn: React.FC<BroadCastTxnProps> = (props) => {
+  const { txn, multisigAccount, chainID } = props;
+  const dispatch = useAppDispatch();
+  const [load, setLoad] = useState(false);
+  const { getChainInfo } = useGetChainInfo();
+  const { rpc, address: walletAddress } = getChainInfo(chainID);
+
+  const updateTxnRes = useAppSelector(
+    (state: RootState) => state.multisig.updateTxnRes
+  );
+
+  useEffect(() => {
+    if (updateTxnRes.status === 'rejected') {
+      dispatch(
+        setError({
+          type: 'error',
+          message: updateTxnRes?.error || FAILED_TO_BROADCAST_ERROR,
+        })
+      );
+    }
+  }, [updateTxnRes]);
+
+  useEffect(() => {
+    return () => {
+      dispatch(resetUpdateTxnState());
+    };
+  }, []);
+
+  const broadcastTxn = async () => {
+    setLoad(true);
+    const authToken = getAuthToken(chainID);
+    const queryParams = {
+      address: walletAddress,
+      signature: authToken?.signature || '',
+    };
+    try {
+      const client = await SigningStargateClient.connect(rpc);
+
+      const multisigAcc = await client.getAccount(
+        multisigAccount?.account.address
+      );
+      if (!multisigAcc) {
+        dispatch(
+          setError({
+            type: 'error',
+            message: 'multisig account does not exist on chain',
+          })
+        );
+        setLoad(false);
+        return;
+      }
+
+      const mapData = multisigAccount.pubkeys || {};
+      let pubkeys: Pubkey[] = [];
+
+      pubkeys = mapData.map((p) => {
+        const parsed = p?.pubkey;
+        const obj = {
+          type: parsed?.type,
+          value: parsed?.value,
+        };
+        return obj;
+      });
+
+      const multisigThresholdPK = NewMultisigThresholdPubkey(
+        pubkeys,
+        `${multisigAccount?.account?.threshold}`
+      );
+
+      const txBody = {
+        typeUrl: '/cosmos.tx.v1beta1.TxBody',
+        value: {
+          messages: txn.messages,
+          memo: txn.memo,
+        },
+      };
+
+      const walletAmino = await getWalletAmino(chainID);
+      const offlineClient = await SigningStargateClient.offline(walletAmino[0]);
+      const txBodyBytes = offlineClient.registry.encode(txBody);
+
+      const signedTx = makeMultisignedTx(
+        multisigThresholdPK,
+        multisigAcc.sequence,
+        txn?.fee,
+        txBodyBytes,
+        new Map(
+          txn?.signatures.map((s) => [s.address, fromBase64(s.signature)])
+        )
+      );
+
+      const result = await client.broadcastTx(
```

Comment:
Move signing and broadcast logic to another file. Let's create a follow up task for this.